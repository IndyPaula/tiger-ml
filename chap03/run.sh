#!/bin/bash

test20s=$(ls ../testcases/test2?.tig | grep -v ../testcases/test24.tig)
test30s=$(ls ../testcases/test3?.tig | grep -v ../testcases/test30.tig)
test40s=$(ls ../testcases/test4?.tig | grep -v ../testcases/test42.tig)

# Explicit call to ml-yacc needed here as there is a bug in CM (Compilation Manager).
ml-yacc tiger.grm &&
  ml-build sources.cm Main.main &&
  head -1 tiger.grm.desc &&
  sml @SMLload=sources.x86-linux \
      *.tig \
      ../testcases/test?.tig \
      ../testcases/test1?.tig \
      ${test20s} \
      ${test30s} \
      ${test40s} \
      > run.actual.out &&
  diff run.expected.out run.actual.out &&
  echo "good!"

echo status=$?
