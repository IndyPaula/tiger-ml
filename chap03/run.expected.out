test1.tig:
test2.tig:
test3.tig:
test4.tig:
test5.tig:
test6.tig:
test7.tig:
test8.tig:
test9.tig:
../testcases/test1.tig:
../testcases/test2.tig:
../testcases/test3.tig:
../testcases/test4.tig:
../testcases/test5.tig:
../testcases/test6.tig:
../testcases/test7.tig:
../testcases/test8.tig:
../testcases/test9.tig:
../testcases/test10.tig:
../testcases/test11.tig:
../testcases/test12.tig:
../testcases/test13.tig:
../testcases/test14.tig:
../testcases/test15.tig:
../testcases/test16.tig:
../testcases/test17.tig:
../testcases/test18.tig:
../testcases/test19.tig:
../testcases/test20.tig:
../testcases/test21.tig:
../testcases/test22.tig:
../testcases/test23.tig:
../testcases/test25.tig:
../testcases/test26.tig:
../testcases/test27.tig:
../testcases/test28.tig:
../testcases/test29.tig:
../testcases/test31.tig:
../testcases/test32.tig:
../testcases/test33.tig:
../testcases/test34.tig:
../testcases/test35.tig:
../testcases/test36.tig:
../testcases/test37.tig:
../testcases/test38.tig:
../testcases/test39.tig:
../testcases/test40.tig:
../testcases/test41.tig:
../testcases/test43.tig:
../testcases/test44.tig:
../testcases/test45.tig:
../testcases/test46.tig:
../testcases/test47.tig:
../testcases/test48.tig:
../testcases/test49.tig:
../testcases/test49.tig:5.18:syntax error: inserting  PLUS
